llvm/include/llvm/Object/ELF.h

```diff
@@ -163,6 +163,49 @@ static inline Error defaultWarningHandler(const Twine &Msg) {
   return createError(Msg);
 }
 
+template <class ELFT>
+static bool checkOffsets(const typename ELFT::Phdr &Phdr,
```

Comment:
Correct me if I'm wrong, but isn't the `static` on a `static` template function in a header pointless?

Reply:
Somewhat. The `static` makes the instantiated functions internal linkage, so you can have another template of the same mangling without ODR violation concern, but that's probably not a good idea anyway.

Dropping static will be better and enable linker duplicate elimination.